src/lightning/pytorch/utilities/_pytree.py

```diff
@@ -0,0 +1,38 @@
+from typing import Any, List, Tuple
+
+from torch.utils._pytree import _get_node_type, LeafSpec, PyTree, SUPPORTED_NODES, tree_unflatten, TreeSpec
+
+
+def _is_leaf(pytree: PyTree) -> bool:
```

Comment:
any dosctring?

Reply:
This is a copy of `torch.utils._pytree._is_leaf` and is not meant to be public. I added comments about the relevant modifications

Reply:
still docstring for future us, or if you want for me :P

Reply:
Done